Scheduled tests failed

static type check failed because the version of pydantic was updated.

https://github.com/hmasdev/runnable_family/actions/runs/10857865500

```bash
$ date +"%Y%m%d"
20240914
$ git clone git@github.com:hmasdev/runnable_family.git
$ cd runnable_family/
$ ls
$ python -m venv .venv
$ source .venv/bin/activate
$ pip install -e .[dev]
(OMISSION)
Successfully installed PyYAML-6.0.2 SQLAlchemy-2.0.34 aiohappyeyeballs-2.4.0 aiohttp-3.10.5 aiosignal-1.3.1 annotated-types-0.7.0 anyio-4.4.0 async-timeout-4.0.3 attrs-24.2.0 autopep8-2.3.1 certifi-2024.8.30 charset-normalizer-3.3.2 coverage-7.6.1 exceptiongroup-1.2.2 flake8-7.1.1 frozenlist-1.4.1 grandalf-0.8 greenlet-3.1.0 h11-0.14.0 httpcore-1.0.5 httpx-0.27.2 idna-3.9 iniconfig-2.0.0 jsonpatch-1.33 jsonpointer-3.0.0 langchain-0.3.0 langchain-core-0.3.0 langchain-text-splitters-0.3.0 langgraph-0.2.21 langgraph-checkpoint-1.0.9 langsmith-0.1.120 mccabe-0.7.0 multidict-6.1.0 mypy-1.11.2 mypy-extensions-1.0.0 numpy-1.26.4 orjson-3.10.7 packaging-24.1 pluggy-1.5.0 pycodestyle-2.12.1 pydantic-2.9.1 pydantic-core-2.23.3 pyflakes-3.2.0 pyparsing-3.1.4 pytest-8.3.3 pytest-cov-5.0.0 pytest-mock-3.14.0 requests-2.32.3 runnable_family-0.0.0a2 sniffio-1.3.1 tenacity-8.5.0 tomli-2.0.1 typing-extensions-4.12.2 urllib3-2.2.3 yarl-1.11.1
$ pip list | grep langchain
langchain                0.3.0
langchain-core           0.3.0
langchain-text-splitters 0.3.0
$ git rev-parse HEAD
2348224c17a8be51012154a28f20b8afa303369c
$ mypy runnable_family
runnable_family/loopback.py:122: error: Argument 1 to "add_node" of "Graph" has incompatible type "type[pydantic.v1.main.BaseModel]"; expected "type[pydantic.main.BaseModel] | Runnable[Any, Any]"  [arg-type]
runnable_family/loopback.py:126: error: Argument 1 to "add_node" of "Graph" has incompatible type "type[pydantic.v1.main.BaseModel]"; expected "type[pydantic.main.BaseModel] | Runnable[Any, Any]"  [arg-type]
Found 2 errors in 1 file (checked 9 source files)
$ pip install -U langchain==0.2.16
(OMISSION)
Successfully installed langchain-0.2.16 langchain-core-0.2.40 langchain-text-splitters-0.2.4
$ mypy runnable_family
Success: no issues found in 9 source files
```

--- a/.github/workflows/code-style-check-workflow-call.yaml
+++ b/.github/workflows/code-style-check-workflow-call.yaml
@@ -40,3 +40,10 @@ jobs:
         run: |
           flake8 runnable_family --show-source
           flake8 tests --show-source
+
+      - name: Run flake8 for langchain<0.3.0
+        run: |
+          # code stylel check for backward compatibility
+          pip install -U "langchain<0.3.0"
+          flake8 runnable_family --show-source
+          flake8 tests --show-source

--- a/.github/workflows/pytest-workflow-call.yaml
+++ b/.github/workflows/pytest-workflow-call.yaml
@@ -39,3 +39,9 @@ jobs:
       - name: Run unit tests
         run: |
           pytest -m 'not integration'
+
+      - name: Run unit tests for langchain<0.3.0
+        run: |
+          # test for backward compatibility
+          pip install -U "langchain<0.3.0"
+          pytest -m 'not integration'

--- a/.github/workflows/static-type-check-workflow-call.yaml
+++ b/.github/workflows/static-type-check-workflow-call.yaml
@@ -40,3 +40,10 @@ jobs:
         run: |
           mypy runnable_family
           mypy tests
+
+      - name: Run mypy for langchain<0.3.0
+        run: |
+          # static type check for backward compatibility
+          pip install -U "langchain<0.3.0"
+          mypy runnable_family
+          mypy tests

--- a/runnable_family/loopback.py
+++ b/runnable_family/loopback.py
@@ -1,4 +1,4 @@
-from langchain.pydantic_v1 import create_model
+import langchain
 from langchain_core.runnables import (
     Runnable,
     RunnableBranch,
@@ -13,6 +13,11 @@ from langgraph.graph.graph import Graph, CompiledGraph, END
 from typing import Callable
 from uuid import uuid4
 from .basic import RunnableConstant, RunnableAdd
+
+if langchain.__version__ < '0.3.0':
+    from langchain.pydantic_v1 import create_model
+else:
+    from pydantic import create_model  # type: ignore
 
 
 class RunnableLoopback(Runnable[Input, Output]):
@@ -119,11 +124,11 @@ class RunnableLoopback(Runnable[Input, Output]):
         try:
             input_node = graph.add_node(self.get_input_schema(config))
         except TypeError:
-            input_node = graph.add_node(create_model(self.get_name("Input")))
+            input_node = graph.add_node(create_model(self.get_name("Input")))  # type: ignore # noqa
         try:
             output_node = graph.add_node(self.get_output_schema(config))
         except TypeError:
-            output_node = graph.add_node(create_model(self.get_name("Output")))
+            output_node = graph.add_node(create_model(self.get_name("Output")))  # type: ignore # noqa
         # cretae sugraphs
         runnable_graph = self._runnable.get_graph(config)
         condition_graph = self._condition.get_graph(config)

